### PR TITLE
Move fork choice rule logic to BFT folder - Closes #4470

### LINKED ISSUE #4470: Move new Fork Choice Rule logic to BFT folder and call it from Block Processor V2 and V1
### Description

We need to move Fork Choice Rule logic (currently in `blocks.js`) to BFT folder `./bft` and call it from `block_processor_v2` and `v1`

### Motivation
Fork choice rule logic deals with BFT properties that better belong to BFT responsibilities so it makes sense to move logic there.
### Acceptance Criteria
Tests are updated

### PR DESCRIPTION
**Pull request by limiaspasdaniel**
### What was the problem?
Refer to #4470 
### How did I solve it?
Refer to #4470
### How to manually test it?
Run `bft.spec.js` with Jest
### Review checklist
- [ ] The PR resolves #4470 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js [hunk 1]
```diff
@@ -725,6 +734,123 @@ describe('bft', () => {
...
+					id: '2',
+					height: 2,
+					version: 2,
+					generatorPublicKey: 'ghijkl',
+					prevotedConfirmedUptoHeight: 1,
+					timestamp: bftInstance.slots.getEpochTime(Date.now()),
+				};
+			});
+
+			it('should return FORK_STATUS_IDENTICAL_BLOCK if isIdenticalBlock evaluates to true', async () => {
```
Comment (ManuGowda) [inline framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js]: do we need async here?

File: framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js [hunk 2]
```diff
@@ -725,6 +734,123 @@ describe('bft', () => {
...
+					...defaults.newBlock,
+					id: defaults.lastBlock.id,
+				};
+
+				expect(bftInstance.forkChoice(aNewBlock, defaults.lastBlock)).toEqual(
+					forkChoiceRule.FORK_STATUS_IDENTICAL_BLOCK,
+				);
+			});
+
+			it('should return FORK_STATUS_VALID_BLOCK if isValidBlock evaluates to true', async () => {
```
Comment (ManuGowda) [inline framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js]: do we need async here?

File: framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js [hunk 3]
```diff
@@ -725,6 +734,123 @@ describe('bft', () => {
...
+						defaults.lastBlock.prevotedConfirmedUptoHeight,
+					height: defaults.lastBlock.height + 1,
+				};
+
+				expect(bftInstance.forkChoice(aNewBlock, defaults.lastBlock)).toEqual(
+					forkChoiceRule.FORK_STATUS_DIFFERENT_CHAIN,
+				);
+			});
+
+			it('should return FORK_STATUS_DISCARD if no conditions are met', async () => {
```
Comment (ManuGowda) [inline framework/test/jest/unit/specs/modules/chain/bft/bft.spec.js]: do we need async here?

File: framework/test/jest/unit/specs/modules/chain/blocks/default_account.js [hunk 4]
```diff
@@ -15,16 +15,16 @@
 'use strict';
 
 const genesisAccount = {
```
Comment (ManuGowda) [inline framework/test/jest/unit/specs/modules/chain/blocks/default_account.js]: maybe format issue? can you revert it back if the format is correct?

Comment (limiaspasdaniel) [reply framework/test/jest/unit/specs/modules/chain/blocks/default_account.js]: This was automatically applied by prettier/eslint so I guess format was wrong before?

File: framework/src/modules/chain/blocks/index.js [hunk 5]
```diff
@@ -27,7 +27,7 @@ const {
 	FORK_STATUS_TIE_BREAK,
 	FORK_STATUS_DIFFERENT_CHAIN,
 	FORK_STATUS_DISCARD,
-} = require('./fork_choice_rule');
+} = require('../bft/fork_choice_rule');
```
Comment (shuse2) [inline framework/src/modules/chain/blocks/index.js]: this export should be in the `bft`

Comment (limiaspasdaniel) [reply framework/src/modules/chain/blocks/index.js]: Updated :)

**Pull request merged**